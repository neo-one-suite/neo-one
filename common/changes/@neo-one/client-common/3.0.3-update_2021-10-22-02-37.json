{
  "changes": [
    {
      "packageName": "@neo-one/client-common",
      "comment": "Update nef model and types",
      "type": "minor"
    }
  ],
  "packageName": "@neo-one/client-common",
  "email": "29364457+spencercorwin@users.noreply.github.com"
}
