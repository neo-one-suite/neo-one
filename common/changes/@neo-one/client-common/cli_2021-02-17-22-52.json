{
  "changes": [
    {
      "packageName": "@neo-one/client-common",
      "comment": "Fix minor bugs.",
      "type": "patch"
    }
  ],
  "packageName": "@neo-one/client-common",
  "email": "spencercorwin@icloud.com"
}
