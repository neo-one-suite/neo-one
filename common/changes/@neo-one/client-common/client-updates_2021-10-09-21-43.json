{
  "changes": [
    {
      "packageName": "@neo-one/client-common",
      "comment": "Add serializeJSON to attributes",
      "type": "patch"
    }
  ],
  "packageName": "@neo-one/client-common",
  "email": "29364457+spencercorwin@users.noreply.github.com"
}
