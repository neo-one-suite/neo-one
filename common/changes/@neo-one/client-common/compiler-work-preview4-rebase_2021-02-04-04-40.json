{
  "changes": [
    {
      "packageName": "@neo-one/client-common",
      "comment": "Update compiler, client, and CLI for Preview4",
      "type": "patch"
    }
  ],
  "packageName": "@neo-one/client-common",
  "email": "spencercorwin@icloud.com"
}
