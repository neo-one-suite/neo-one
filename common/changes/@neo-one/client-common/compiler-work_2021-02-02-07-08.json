{
  "changes": [
    {
      "comment": "Update for Preview3, primarily in compiler, client, and CLI.",
      "type": "patch",
      "packageName": "@neo-one/client-common"
    }
  ],
  "packageName": "@neo-one/client-common",
  "email": "spencercorwin@icloud.com"
}
