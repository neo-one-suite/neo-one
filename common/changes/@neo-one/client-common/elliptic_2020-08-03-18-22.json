{
  "changes": [
    {
      "packageName": "@neo-one/client-common",
      "comment": "Update elliptic dependency.",
      "type": "none"
    }
  ],
  "packageName": "@neo-one/client-common",
  "email": "spencercorwin@icloud.com"
}