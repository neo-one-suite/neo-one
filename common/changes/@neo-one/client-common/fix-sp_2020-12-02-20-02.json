{
  "changes": [
    {
      "packageName": "@neo-one/client-common",
      "comment": "Fix synchronized-promise in browser.",
      "type": "minor"
    }
  ],
  "packageName": "@neo-one/client-common",
  "email": "spencercorwin@icloud.com"
}
