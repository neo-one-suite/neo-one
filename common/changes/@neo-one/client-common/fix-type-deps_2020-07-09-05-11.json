{
  "changes": [
    {
      "packageName": "@neo-one/client-common",
      "comment": "Add appropriate @types dependencies.",
      "type": "patch"
    }
  ],
  "packageName": "@neo-one/client-common",
  "email": "spencercorwin@icloud.com"
}