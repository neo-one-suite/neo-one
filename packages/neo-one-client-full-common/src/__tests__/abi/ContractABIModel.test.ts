import { ContractParameterTypeModel } from '@neo-one/client-common';
import {
  contractAbiModel,
  contractEventDescriptorModel,
  contractMethodDescriptorModel,
  contractParamDefinitionModel,
} from '../../__data__';

describe('ContractABIModel - serializeJSON', () => {
  test('Simple', () => {
    const json = contractAbiModel().serializeJSON();
    expect(json).toMatchSnapshot();
  });

  test('Multiple', () => {
    const json = contractAbiModel(
      [
        contractMethodDescriptorModel([contractParamDefinitionModel.integer, contractParamDefinitionModel.hash160]),
        contractMethodDescriptorModel([contractParamDefinitionModel.string], ContractParameterTypeModel.Integer),
      ],
      [contractEventDescriptorModel([contractParamDefinitionModel.hash160, contractParamDefinitionModel.hash256])],
    ).serializeJSON();
    expect(json).toMatchSnapshot();
  });
});
