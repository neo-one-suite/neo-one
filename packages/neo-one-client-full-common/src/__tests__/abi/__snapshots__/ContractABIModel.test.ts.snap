// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContractABIModel - serializeJSON Multiple 1`] = `
Object {
  "events": Array [
    Object {
      "name": "event",
      "parameters": Array [
        Object {
          "name": "param",
          "type": "Hash160",
        },
        Object {
          "name": "param",
          "type": "Hash256",
        },
      ],
    },
  ],
  "hash": "0x0101010101010101010101010101010101010101",
  "methods": Array [
    Object {
      "name": "function",
      "offset": undefined,
      "parameters": Array [
        Object {
          "name": "param",
          "type": "Integer",
        },
        Object {
          "name": "param",
          "type": "Hash160",
        },
      ],
      "returnType": "Void",
    },
    Object {
      "name": "function",
      "offset": undefined,
      "parameters": Array [
        Object {
          "name": "param",
          "type": "String",
        },
      ],
      "returnType": "Integer",
    },
  ],
}
`;

exports[`ContractABIModel - serializeJSON Simple 1`] = `
Object {
  "events": Array [
    Object {
      "name": "event",
      "parameters": Array [
        Object {
          "name": "param",
          "type": "Boolean",
        },
      ],
    },
  ],
  "hash": "0x0101010101010101010101010101010101010101",
  "methods": Array [
    Object {
      "name": "function",
      "offset": undefined,
      "parameters": Array [
        Object {
          "name": "param",
          "type": "Boolean",
        },
      ],
      "returnType": "Void",
    },
  ],
}
`;
